### PR TITLE
Implement way for SaveACopyAs to open the copy post-save

### LINKED ISSUE #11861: [Feature Request] Save a Copy As and then OPEN THE COPY
Often I need to copy a file that I have open in Notepad++ and then edit that file immediately, while keeping the original file open as well (for reference or independent change).

This becomes an operation of multiple steps because neither _Save As..._ or _Save a Copy As..._ will conclude with both the original and the copy open in Notepad++ tabs.

So I would like to see a way to achieve this in one step, e.g. a new command called _Save a Copy As and Open The Copy..._.

### Debug Information

Above comments based on behavior current to Notepad++ 8.4.2

### Workarounds

Currently I work-around the (minor) pain of the missing feature by:

* Using _Save As_ and then _File Open_ and then find my original file and reopen it
* Using _Save a Copy As_ and then _File Open_ and then find my new file and open it
* Getting an Exporer window in the parent folder of the original file, use drag-n-drop copy of the file, rename the file, and then drag-n-drop the renamed file into Notepad++

### PR DESCRIPTION
**Pull request by alankilborn**
Fix #11861

Note, the design of this PR is based on some comments in a previously abandoned PR; see https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14742#issuecomment-2087640927

### CODE REVIEW

**Comment by dfs-** [comment]
I liked the initial idea after sometimes having the same workflow as described in the issue. That SHIFT+Click _hidden feature_ is a nice touch, as it does not interfere with other users workflow.

**Comment by alankilborn** [comment]
I should have included a visual:

image

---

Note that it also works without the mouse being involved, as the _Save_ button is the default button, so holding Shift while pressing Enter also actuates the proposed new functionality.

File: PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp [hunk 1]
```diff
@@ -902,6 +927,8 @@ class CustomFileDialog::Impl
 	bool _enableFileTypeCheckbox = false;
 	bool _fileTypeCheckboxValue = false;	// initial value
 	wstring _fileTypeCheckboxLabel;
+	bool _isSavingAsCopy = false;
+	bool _isOpeningTheCopy = false;
```
Comment (donho) [inline PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp:931]: IMO, for this feature we need only 1 member variable than 2 bool _isSavingAsCopy & _isOpeningTheCopy, for the reason to make the class class `CustomFileDialog::Impl` simpler.
The 2 bool member variables can be reduced indeed to 1 unsigned char member variable, for 1 feature.
```cpp
unsigned char _savingAsCopyInfo = 0;
```

and use mask SAVE_AS_COPY_DLG (value 0x02) and mask SAVE_AS_COPY_OPEN (value 0x01) to retrieve these 2 booleans when it's necessary.

**Review by donho — CHANGES_REQUESTED** [review]
Squeeze 2 member variables in 1 to make the concerning class interface more trivial & intuitive.

File: PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp (hunk 1 above)
Comment (alankilborn) [reply PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp:931]: OK, I will do it when I return from my most recent trip (extends through 21 July).

Comment (donho) [reply PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp:931]: Bon voyage @alankilborn !

**Comment by alankilborn** [comment]
@donho 's "Merge branch 'master' into i11861b" messed me up somehow; I'll have to close this and create a new PR.

**Pull request closed without merging**